src/main/resources/assets/tconstruct/models/block/smeltery/casting/table.json

```diff
@@ -0,0 +1,110 @@
+{
+    "parent": "block/block",
+    "textures": {
+        "top": "tconstruct:block/smeltery/casting/castingtable_top",
```

Comment:
You are in the casting folder, just `casting/table_top` will suffice. I hate the old 1.12 names, but did not want to make a breaking change for resource packs.

Reply:
Aight, since we're breaking and fixing everything already may as well
do it.